### PR TITLE
update JVMChaos document

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
Signed-off-by: xiang <xiang13225080@163.com>

close issue https://github.com/chaos-mesh/website/issues/162

### CODE REVIEW

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 1]
```diff
@@ -2,185 +2,231 @@
...
+| class     |  string 类型        | Java 类的名称    | 是       |
+| method      | string 类型           | 方法名称      | 是       |
+| exception | string 类型 | 抛出的自定义异常，例如：'java.io.IOException("BOOM")'   | 是       |
+| port   | int 类型     | 附加到 Java 进程 agent 的端口号，通过该端口号将故障注入到 Java 进程 | 否       |
+
+#### stress
+
+| 参数      | 类型                    | 说明                     | 是否必填 |
+| --------- | ----------------------- | ------------------------ | -------- |
+| cpu-count     |  int 类型        | 增加 CPU 压力所使用的 CPU 核的数量，`cpu-count` 和 `mem-size` 中必须配置一个    | 否       |
```
Comment (g1eny0ung) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:202]: Need to convert to `camelCase`.

Comment (WangXiangUSTC) [reply i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md]: Yeah, forget to update. Done

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 2]
```diff
@@ -2,185 +2,231 @@
 title: 模拟 JVM 应用故障
 ---
 
-## JVMChaos 介绍
+Chaos Mesh 通过 [Byteman](https://github.com/chaos-mesh/byteman) 模拟 JVM 应用故障，主要支持以下几种故障类型：
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:5]:
```suggestion
Chaos Mesh 通过 [Byteman](https://github.com/chaos-mesh/byteman) 模拟 JVM 应用故障，主要支持以下类型的故障：：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 3]
```diff
@@ -2,185 +2,231 @@
...
-目前 Chaos Mesh 使用 [MutatingAdmissionWebhook](https://kubernetes.io/zh/docs/reference/access-authn-authz/admission-controllers/#mutatingadmissionwebhook) 修改对 Pod 的定义，通过 [Init 容器](https://kubernetes.io/zh/docs/concepts/workloads/pods/init-containers/)加载 java agnet，并非运行时加载 java agent。因此在使用时存在如下限制：
+1. 单击实验页面中的“新的实验”按钮创建实验：
 
-- Kubernetes 需要启用 Webhook 支持。
-- 在为命名空间配置 MutatingAdmissionWebhook 之前已经存在 Pod，不会受到 JVMChaos 影响。
-- 命名空间下的所有容器中的 JVM 都会在启动阶段加载 java agent，JVMChaos 在被删除后也不会卸载 java agent。若考虑到 java agent 可能对程序行为或性能带来的影响，期望清理 java agnet，请将工作负载移出该命名空间。
+   ![创建实验](./img/create-new-exp.png)
 
-另外，目前无法通过 Chaos Dashboard 创建 JVMChaos。
+2. 在“选择目标”处选择 “JVM 故障”，然后选择具体行为，例如 `RETURN`，最后填写具体配置：
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:22]:
```suggestion
2. 在“选择目标”处选择 “JVM 故障”，然后选择具体行为（如 `RETURN`），最后填写具体配置：
```

Comment (WangXiangUSTC) [reply i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md]: done

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md (hunk 2 above)
Comment (WangXiangUSTC) [reply i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md]: done

**Comment by WangXiangUSTC** [comment]
@en-jin19 PTAL again

**Comment by WangXiangUSTC** [comment]
@STRRL PTAL

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 4]
```diff
@@ -2,185 +2,231 @@
...
+## 使用 YAML 方式创建实验
 
-```shell
-kubectl apply -f sidecar-template.yaml
-kubectl apply -f sidecar.yaml
-```
+下面将以指定返回值为例，展示 JVMChaos 的使用方法与效果。以下内容中涉及的 yaml 文件均可在 [examples/jvm](https://github.com/chaos-mesh/chaos-mesh/tree/master/examples/jvm) 中找到，以下步骤默认的工作路径也是在 `examples/jvm` 中。 默认 Chaos Mesh 安装的命名空间为 `chaos-testing`。
 
-### 2. 创建被测应用
+### 1. 创建被测应用
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:38]:
```suggestion
### 第 1 步：创建被测应用
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 5]
```diff
@@ -2,185 +2,231 @@
...
+      # this application will print log like this below:
+      # 0. Hello World
+      # 1. Hello World
+      # ...
+      image: xiang13225080/helloworld:v1.0
+      imagePullPolicy: IfNotPresent
 ```
 
-其中值为 `admission-webhook.chaos-mesh.org/request: jvmchaos-sidecar` 的 `annotation` 与步骤 1 `sidecar.yaml` 中 `ConfigMap` 的名称对应。
+创建应用所属的 namespace：
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:60]:
```suggestion
1. 创建应用所属的 namespace：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 6]
```diff
@@ -2,185 +2,231 @@
...
+
+```shell
+kubectl create namespace helloworld
+```
 
-建立应用 Deployment：
+建立该应用 Pod：
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:66]:
```suggestion
2. 建立该应用 Pod：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 7]
```diff
@@ -2,185 +2,231 @@
...
 
 ```shell
 kubectl apply -f app.yaml
 ```
 
-执行 `kubectl -n app get pods`，预期能够观察到命名空间 `app` 中出现 `1` 个名称形如 `springboot-jvmchaos-demo-777d94c5b9-7t7l2` 的 Pod，等待其 `READY` 为 `1/1` 后进行下一步。
+执行 `kubectl -n helloworld get pods`，预期能够观察到命名空间 `helloworld` 中名为 `helloworld` 的 Pod，等待其 `READY` 为 `1/1` 后进行下一步。
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:72]:
```suggestion
3. 执行 `kubectl -n helloworld get pods`，预期能够观察到命名空间 `helloworld` 中名为 `helloworld` 的 Pod，等待其 `READY` 为 `1/1` 后进行下一步。
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 8]
```diff
@@ -2,185 +2,231 @@
...
-springboot-jvmchaos-demo-777d94c5b9-7t7l2   1/1     Running   0          21s
+kubectl get pods -n helloworld            
+NAME         READY   STATUS    RESTARTS   AGE
+helloworld   1/1     Running   0          2m
 ```
 
-### 3. 观测未被注入时的行为
-
-在注入前你可以先观测应用 `jvm-chaos-demo` 未被注入时的行为，例如：
+### 2. 观测未被注入时的行为
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:86]:
```suggestion
### 第 2 步：观测未被注入时的行为
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 9]
```diff
@@ -2,185 +2,231 @@
...
+2. Hello World
+3. Hello World
+4. Hello World
+5. Hello World
 ```
 
-### 4. 注入 JVMChaos 并验证
+可以看到 `helloworld` 每隔一秒输出一行 `Hello World`，每行的编号依次递增。
+
+### 3. 注入 JVMChaos 并验证
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:107]:
```suggestion
### 第 3 步：注入 JVMChaos 并验证
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 10]
```diff
@@ -2,185 +2,231 @@
...
+日志如下所示：
+
+```shell
+Rule.execute called for return_0:0
+return execute
+caught ReturnException
+9999. Hello World
+```
+
+### 字段说明
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:151]:
```suggestion
## 字段说明
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 11]
```diff
@@ -2,185 +2,231 @@
...
+| latency  | 增加方法调用延迟                       |
+| return | 修改方法返回值                             |
+| exception | 抛出自定义异常           |
+| stress    | 提高 java 进程 CPU 使用率，或者造成       内存溢出（支持堆、栈溢出）    |
+| gc    | 触发垃圾回收 |
+| ruleData    | 设置 Byteman 配置触发故障 |
+
+针对不同的 `action`，有不同的配置项可以填写。
+
+#### latency
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:171]:
```suggestion
### Latency
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 12]
```diff
@@ -2,185 +2,231 @@
...
+
+| 参数      | 类型                    | 说明                     | 是否必填 |
+| --------- | ----------------------- | ------------------------ | -------- |
+| class     |  string 类型        | Java 类的名称    | 是       |
+| method      | string 类型           | 方法名称      | 是       |
+| latency | int 类型 | 增加方法的延迟时间，单位为 ms    | 是       |
+| port   | int 类型     | 附加到 Java 进程 agent 的端口号，通过该端口号将故障注入到 Java 进程 | 否       |
+
+#### return
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:180]:
```suggestion
### Return 相关参数
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 13]
```diff
@@ -2,185 +2,231 @@
...
+
+| 参数      | 类型                    | 说明                     | 是否必填 |
+| --------- | ----------------------- | ------------------------ | -------- |
+| class     |  string 类型        | Java 类的名称    | 是       |
+| method      | string 类型           | 方法名称      | 是       |
+| value | string 类型 | 指定方法的返回值，目前支持数字和字符串类型的返回值，如果为字符串，则需要使用双引号，例如："chaos"。    | 是       |
+| port   | int 类型     | 附加到 Java 进程 agent 的端口号，通过该端口号将故障注入到 Java 进程 | 否       |
+
+#### exception
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:189]:
```suggestion
### Exception 相关参数
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 14]
```diff
@@ -2,185 +2,231 @@
...
+
+| 参数      | 类型                    | 说明                     | 是否必填 |
+| --------- | ----------------------- | ------------------------ | -------- |
+| class     |  string 类型        | Java 类的名称    | 是       |
+| method      | string 类型           | 方法名称      | 是       |
+| exception | string 类型 | 抛出的自定义异常，例如：'java.io.IOException("BOOM")'   | 是       |
+| port   | int 类型     | 附加到 Java 进程 agent 的端口号，通过该端口号将故障注入到 Java 进程 | 否       |
+
+#### stress
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:198]:
```suggestion
### Stress 相关参数
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 15]
```diff
@@ -2,185 +2,231 @@
...
+
+| 参数      | 类型                    | 说明                     | 是否必填 |
+| --------- | ----------------------- | ------------------------ | -------- |
+| cpuCount     |  int 类型        | 增加 CPU 压力所使用的 CPU 核的数量，`cpuCount` 和 `memType` 中必须配置一个    | 否       |
+| memType      | string 类型           | 内存 OOM 的类型，目前支持 "stack" 和 "heap" 两种类型      | 无       |
+| port   | int 类型     | 附加到 Java 进程 agent 的端口号，通过该端口号将故障注入到 Java 进程 | 否       |
+
+#### gc
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:206]:
```suggestion
### GC 相关参数
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 16]
```diff
@@ -2,185 +2,231 @@
...
+
+| 参数      | 类型                    | 说明                     | 是否必填 |
+| --------- | ----------------------- | ------------------------ | -------- |
+| port   | int 类型     | 附加到 Java 进程 agent 的端口号，通过该端口号将故障注入到 Java 进程 | 否       |
+
+#### ruleData
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:212]:
```suggestion
### RuleData 相关参数
```

**Comment by en-jin19** [comment]
@WangXiangUSTC I have left some comments, PTAL, thanks!

**Comment by WangXiangUSTC** [comment]
@en-jin19 All comments addressed, thanks! PTAL again

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 17]
```diff
@@ -2,185 +2,231 @@
...
+CLASS Main
+METHOD getnum
+AT ENTRY
+IF true
+DO
+    return 9999
+ENDRULE
+```
+
+将配置中的换行转换为换行符 "\n"，将转换后的数据设置为参数 "ruleData" 的值，如上的配置转换为 "\nRULE modify return value\nCLASS Main\nMETHOD getnum\nAT ENTRY\nIF true\nDO return 9999\nENDRULE\n"。
```
Comment (STRRL) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:232]: 我比较倾向于用代码段替换双引号， PTAL

rest LGTM

Comment (WangXiangUSTC) [reply i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md]: done

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 18]
```diff
@@ -2,185 +2,235 @@
...
+4. Hello World
+5. Hello World
 ```
 
-### 4. 注入 JVMChaos 并验证
+可以看到 `helloworld` 每隔一秒输出一行 `Hello World`，每行的编号依次递增。
+
+### 第 3 步：注入 JVMChaos 并验证
 
 指定返回值的 JVMChaos 内容如下：
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:109]:
```suggestion
1. 指定返回值的 JVMChaos 内容如下：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 19]
```diff
@@ -2,185 +2,235 @@
...
-    labelSelectors:
-      app: springboot-jvmchaos-demo
+    namespaces:
+      - helloworld
 ```
 
-JVMChaos 将 `hello` 方法的返回值修改为字符串 `hello chaos mesh!`。
+JVMChaos 将 `getnum` 方法的返回值修改为数字 `9999`，也就是让 `helloworld` 的每行输出的编号都设置为 `9999`。
 
 注入指定返回值的 JVMChaos：
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:130]:
```suggestion
2. 注入指定返回值的 JVMChaos：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 20]
```diff
@@ -2,185 +2,235 @@
...
 
 ```shell
 kubectl apply -f ./jvm-return-example.yaml
 ```
 
-使用 curl 或者直接使用浏览器访问 <http://localhost:8080/hello>，预期返回 `hello chaos mesh!`：
+查看 helloworld 的最新日志：
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:136]:
```suggestion
3. 查看 helloworld 的最新日志：
```

**Review by en-jin19 — APPROVED** [review]
Rest LGTM

File: i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md [hunk 21]
```diff
@@ -2,185 +2,235 @@
...
-```
-
-为 JVMChaos 所需要的修改行为准备模板：
+## 使用 YAML 方式创建实验
 
-```shell
-kubectl apply -f sidecar-template.yaml
-kubectl apply -f sidecar.yaml
-```
+下面将以指定返回值为例，展示 JVMChaos 的使用方法与效果。以下内容中涉及的 yaml 文件均可在 [examples/jvm](https://github.com/chaos-mesh/chaos-mesh/tree/master/examples/jvm) 中找到，以下步骤默认的工作路径也是在 `examples/jvm` 中。 默认 Chaos Mesh 安装的命名空间为 `chaos-testing`。
```
Comment (en-jin19) [inline i18n/zh/docusaurus-plugin-content-docs/current/simulate-jvm-application-chaos.md:36]:
```suggestion
下面将以指定返回值为例，展示 JVMChaos 的使用方法与效果。以下内容中涉及的 YAML 文件均可在 [examples/jvm](https://github.com/chaos-mesh/chaos-mesh/tree/master/examples/jvm) 中找到，以下步骤默认的工作路径也是在 `examples/jvm` 中。 默认 Chaos Mesh 安装的命名空间为 `chaos-testing`。
```

**Comment by WangXiangUSTC** [comment]
@en-jin19 Thanks! PTAL again

**Review by cwen0 — APPROVED** [review]
LGTM

**Pull request merged**